contract/src/main.rs

```diff
@@ -2137,43 +1897,14 @@ pub extern "C" fn migrate() {
         .unwrap_or(EventsMode::NoEvents);
 
     if let Some(optional_events_mode) = optional_events_mode {
-        let requested_events_mode: EventsMode = optional_events_mode.try_into().unwrap_or_revert();
-        match (current_events_mode, requested_events_mode) {
-            (EventsMode::CES, EventsMode::CES) => casper_event_standard::emit(Migration::new()),
-            (_, EventsMode::CES) => {
-                // Initialize events structures.
-                utils::init_events();
-                casper_event_standard::emit(Migration::new());
-            }
-            (_, EventsMode::CEP47) => record_cep47_event_dictionary(CEP47Event::Migrate),
-            (_, EventsMode::Native) => {}
-            (_, EventsMode::Native47) => {}
-            (_, EventsMode::NativeNCES) => {
+        if optional_events_mode as u8 != current_events_mode as u8 {
+            runtime::put_key(EVENTS_MODE, storage::new_uref(optional_events_mode).into());
+            if let Some(_) = get_key(casper_event_standard::EVENTS_DICT) {
```

Comment:
Same as cep-18, I don't get why it is not a None here. If the dict does not exist then init_events ?